docs/DEVELOPING.md

```diff
@@ -155,53 +155,87 @@ the collator node the parachain will be running on.
 ./scripts/runtime_benchmarks.sh <runtime>
 ```
 
-# Upgrading to a newer Polkadot / Substrate / Cumulus version
+# Updating to a newer version of Polkadot
 
-When a new version of Polkadot is released, a mirroring release happens for the other
-parity projects such as Substrate and Cumulus, but also for other third-party projects
+When a new version of Polkadot is released, companion releases happen for the other
+parity projects such as Substrate and Cumulus, as well as for other third-party projects
 such as the `orml` pallets, `xcm-simulator`, etc.
 
-You can follow these steps tailored to update only one of those specific projects
-updating the three simultaneously is the most common flow.
+Therefore, updating this repository to a new version of Polkadot means updating all of these dependencies
+(internal and external to Centrifuge ) and have them all aligned on the same version of Polkadot.
 
-The flow to upgrade to a newer version of polkadot is usually as follows:
+_Note: When we say "new version of Polkadot", we implicitly mean "Polkadot, Substrate, Cumulus"._
 
-1. Use [`diener`](https://github.com/bkchr/diener) to update Polkadot, Substrate, and Cumulus to said release
+The high level flow to upgrade to a newer version of Polkadot is:
 
-```shell
-export POLKADOT_NEW_VERSION="<version>"; # for example, 0.9.24
+1. Update all the Centrifuge-Chain dependencies to a revision that also points to the last version of Polkadot 
+2. Fix all the breaking changes introduced by the latest versions
 
-diener update --polkadot --branch release-v$POLKADOT_NEW_VERSION;
-diener update --substrate --branch polkadot-v$POLKADOT_NEW_VERSION;
-diener update --cumulus --branch polkadot-v$POLKADOT_NEW_VERSION;
-```
 
-2. Repeat step 1 for our other repos that we depend on:
+### 1. Update dependencies 
 
-- [centrifuge/chainbridge-substrate](https://github.com/centrifuge/chainbridge-substrate)
-- [centrifuge/unique-assets](https://github.com/centrifuge/unique-assets)
-- [centrifuge/fudge](https://github.com/centrifuge/fudge)
+0. **Update the `cargo patch` rules in `Cargo.toml`**
 
-3. Make sure that we point to the latest revision of the projects altered in step 2
+    The cargo patch rules ensure that we use specific revision for Substrate, Polkadot, Cumulus, and others, by
+    pointing to a specific git revision. For each of the projects covered by these rules, look up the latest git 
+    revision for the new release and find-replace all in the root `Cargo.toml` file.
```

Comment:
Since it is a fairly deterministic step, maybe we can add this command (to be repeated for every rev hash that needs update:
```
sed -i 's/$old_rev/$new_rev/g' Cargo.toml
```

Reply:
I guess everyone will have a different preferred method of doing this so I will skip this one. For me it's easier to open `Cargo.toml` on clion, get the old revision and do a find-replace-all right there, for example. But I appreciate it 👍